server/src/main/java/org/elasticsearch/action/search/TransportSearchAction.java

```diff
@@ -484,11 +485,66 @@ void executeRequest(
                 }
             }
         });
+        final SearchSourceBuilder source = original.source();
```

Comment:
It is not entirely clear to me if this specific change (the introduction of open/close PIT) is required for this specific change. Could it be extracted into a separate change? As far as I understand it is needed only once we introduce compound retrievers?

Reply:
See my previous comments, I opened another pr to focus this PR on introducing a compound retriever mode in the search action. I'll update this PR once we agree on the other one ;).